### DISCUSSION TITLE
Release 4.2

### DISCUSSION THREAD

**Discussion by revoxhere** [revoxhere/duino-coin#1807] [General]
## What's new, what's changed...
 First of all, **Duino BlushyBox is available again for purchase in new vibrant colours :orange_heart: :green_heart: :blue_heart: :yellow_heart:** and an optional **DS18B20** temperature sensor for Duino IoT. Check it out at [starter.duinocoin.com](https://starter.duinocoin.com)
* ESP Code
  * 128x64 SSD1306 OLED support for displaying mining info by @revoxhere 
  * 16x2 LCD support for displaying mining info by @revoxhere 
  * Hardware abstraction layer for adding more displays in the future by @revoxhere 
  * DNS setting for ESP32 by @revoxhere 
  * HSU07M sensor support by [omarherrera1978](https://www.reddit.com/user/omarherrera1978/) [suggested [on Reddit](https://www.reddit.com/r/Duinocoin/comments/1e3fsdp/hsu07m_sensor_test_with_esp8266_using_the_code/)]
  * Improved connection wait logic for more reliable network by @revoxhere 
  * Captive portal (AP mode) option by @revoxhere 
  * BlushyBox support by @revoxhere 
  * Ability to disable the brownout detector for ESP32 by @revoxhere 
  * ESP8266 watchdog timer for preventing freezes by @revoxhere 
  * Improved code logic to prevent ESP8266 from randomly freezing by @revoxhere 
  * Improved ESP32 task handling for huge performance boost (up to 190 kH/s on dual core variants) by @revoxhere 
  * Fixed web dashboard not working on dual core ESP32 variants by @revoxhere 
  * Baudrate setting option by @revoxhere

* PC Miner and AVR Miner
  * Only check and ask for a mining key if it's enabled by @revoxhere

* Other
  * Update duco-install-rpi.sh by @ZL1LAC in https://github.com/revoxhere/duino-coin/pull/1789
  * Update README.md by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1803
  * Update README.md, added some links to useful packages  by @mavotronik in https://github.com/revoxhere/duino-coin/pull/1804

## New Contributors - welcome!
* @mavotronik made their first contribution in https://github.com/revoxhere/duino-coin/pull/1804

### ⚠️ NOTICE: .exe binaries may be detected as a virus by your antivirus software. This is a **false positive** caused by pyinstallers bootloader, read about it [here](https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg). Duino-Coin is not a virus.

**Full Changelog**: https://github.com/revoxhere/duino-coin/compare/4.1...4.2
As always, please report any encountered issues. Happy mining!

<hr /><em>This discussion was created from the release <a href='https://github.com/revoxhere/duino-coin/releases/tag/4.2'>Release 4.2</a>.</em>

**Comment by havuong19797** [comment]
How to use 128x64 SSD1306 OLED
